roles/st2/tasks/main.yml

```diff
@@ -1,17 +1,25 @@
 ---
-- name: Install latest st2 package
+- name: Install libffi-devel on EL6
   become: yes
-  apt:
+  package:
+    name: ftp://fr2.rpmfind.net/linux/centos/6/os/x86_64/Packages/libffi-devel-3.0.5-3.2.el6.x86_64.rpm
+    state: latest
+  when: ansible_os_family == "RedHat" and ansible_distribution_major_version == "6"
+  ignore_errors: yes  # Ignore error when already installed
```

Comment:
Additionally, this task does run idempotently, now that I've changed to `state: latest`:

```
TASK [st2 : Install libffi-devel on EL6] ***************************************
ok: [10.0.4.91]
```

This should also make the `ignore_errors` statement pointless, so I have removed it in 100ca71. So as long as you don't prefer the `server-optional` approach (and know what I'm doing wrong that is preventing me from installing `libffi-devel` that way) then this should work better.